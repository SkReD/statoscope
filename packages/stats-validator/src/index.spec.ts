import Validator from './';

test.skip('should work', async () => {
  const validator = new Validator({
    plugins: [[require.resolve('../../stats-validator-plugin-webpack'), 'webpack']],
    rules: {
      'webpack/restricted-modules': ['error', [/\/src\//]],
      'webpack/restricted-packages': ['error', ['foo']],
    },
  });

  const result = await validator.validate(
    require.resolve('../../../test/bundles/v5/simple/stats-prod.json'),
  );

  result.files.input = result.files.input.replace(process.cwd(), '<pwd>');
  expect(result).toMatchSnapshot();
});

test.skip('custom reporter', async () => {
  const validator = new Validator({
    plugins: [[require.resolve('../../stats-validator-plugin-webpack'), 'webpack']],
    reporters: [
      [
        require.resolve('../../../test/fixtures/stats-validator/reporters/custom.js'),
        'foo',
      ],
    ],
    rules: {
      'webpack/restricted-modules': ['error', [/\/src\//]],
      'webpack/restricted-packages': ['error', ['foo']],
    },
  });

  const result = await validator.validate(
    require.resolve('../../../test/bundles/v5/simple/stats-prod.json'),
  );

  result.files.input = result.files.input.replace(process.cwd(), '<pwd>');
  expect(result).toMatchSnapshot();
});

test.skip('silent', async () => {
  const validator = new Validator({
    plugins: [[require.resolve('../../stats-validator-plugin-webpack'), 'webpack']],
    reporters: [
      [
        require.resolve('../../../test/fixtures/stats-validator/reporters/custom.js'),
        'foo',
      ],
    ],
    rules: {
      'webpack/restricted-modules': ['error', [/\/src\//]],
      'webpack/restricted-packages': ['error', ['foo']],
    },
  });

  const result = await validator.validate(
    require.resolve('../../../test/bundles/v5/simple/stats-prod.json'),
  );

  result.files.input = result.files.input.replace(process.cwd(), '<pwd>');
  expect(result).toMatchSnapshot();
});
