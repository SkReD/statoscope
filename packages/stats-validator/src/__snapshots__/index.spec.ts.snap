// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`custom reporter 1`] = `
{
  "files": {
    "input": "<pwd>/test/bundles/v5/simple/stats-prod.json",
    "reference": undefined,
  },
  "rules": [
    {
      "api": {
        "getRuleDescriptor": [Function],
        "getStorage": [Function],
        "message": [Function],
        "setRuleDescriptor": [Function],
      },
      "execParams": {
        "mode": "error",
      },
      "name": "webpack/restricted-modules",
    },
    {
      "api": {
        "getRuleDescriptor": [Function],
        "getStorage": [Function],
        "message": [Function],
        "setRuleDescriptor": [Function],
      },
      "execParams": {
        "mode": "error",
      },
      "name": "webpack/restricted-packages",
    },
  ],
}
`;

exports[`should work 1`] = `
{
  "files": {
    "input": "<pwd>/test/bundles/v5/simple/stats-prod.json",
    "reference": undefined,
  },
  "rules": [
    {
      "api": {
        "getRuleDescriptor": [Function],
        "getStorage": [Function],
        "message": [Function],
        "setRuleDescriptor": [Function],
      },
      "execParams": {
        "mode": "error",
      },
      "name": "webpack/restricted-modules",
    },
    {
      "api": {
        "getRuleDescriptor": [Function],
        "getStorage": [Function],
        "message": [Function],
        "setRuleDescriptor": [Function],
      },
      "execParams": {
        "mode": "error",
      },
      "name": "webpack/restricted-packages",
    },
  ],
}
`;

exports[`silent 1`] = `
{
  "files": {
    "input": "<pwd>/test/bundles/v5/simple/stats-prod.json",
    "reference": undefined,
  },
  "rules": [
    {
      "api": {
        "getRuleDescriptor": [Function],
        "getStorage": [Function],
        "message": [Function],
        "setRuleDescriptor": [Function],
      },
      "execParams": {
        "mode": "error",
      },
      "name": "webpack/restricted-modules",
    },
    {
      "api": {
        "getRuleDescriptor": [Function],
        "getStorage": [Function],
        "message": [Function],
        "setRuleDescriptor": [Function],
      },
      "execParams": {
        "mode": "error",
      },
      "name": "webpack/restricted-packages",
    },
  ],
}
`;
